src/main/java/org/opentripplanner/middleware/models/MonitoredComponent.java

```diff
@@ -0,0 +1,88 @@
+package org.opentripplanner.middleware.models;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.google.common.collect.Maps;
+import com.mongodb.client.model.Filters;
+import org.bson.codecs.pojo.annotations.BsonIgnore;
+import org.bson.conversions.Bson;
+import org.opentripplanner.middleware.auth.RequestingUser;
+import org.opentripplanner.middleware.persistence.Persistence;
+import org.opentripplanner.middleware.utils.ConfigUtils;
+import org.opentripplanner.middleware.utils.JsonUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * This class represents the system components (e.g., servers or UIs) that otp-middleware is expected to monitor. It
+ * should serve as a place to collect properties needed for integrations with various monitoring services (e.g.,
+ * Bugsnag).
+ */
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class MonitoredComponent extends Model {
+    private static final Logger LOG = LoggerFactory.getLogger(MonitoredComponent.class);
+    /**
+     * The field that maps this component to a project in Bugsnag (this is a string UUID).
+     *
+     * See https://api.bugsnag.com/organizations/<organization_id>/projects for a list of projects.
+     */
+    public String bugsnagProjectId;
+    /**
+     * The name of the component (e.g., datatools-server) to display to otp-admin-ui users.
+     */
+    public String name;
+
+    /**
+     * Read in the list of {@link MonitoredComponent} from the MONITORED_COMPONENTS field in the env.yml file.
+     */
+    public static void initializeMonitoredComponentsFromConfig() {
+        List<MonitoredComponent> configComponents;
+        try {
+            configComponents = JsonUtils.getPOJOFromJSONAsList(
+                ConfigUtils.getConfigProperty("MONITORED_COMPONENTS"),
+                MonitoredComponent.class
+            );
+            if (configComponents == null) {
+                throw new IllegalArgumentException("MONITORED_COMPONENTS should not be missing.");
+            }
```

Comment:
It is optional in the sense that the application doesn't fail if it's missing. This exception is thrown so that a null value will be caught in the same way that a parsing error were to happen.

Reply:
What happens if someone does not enter a `MONITORED_COMPONENTS` in the config file?